File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/ResumeUserRequestDto.java
package com.ungs.docsys.dtos;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ResumeUserRequestDto {
    @NotNull(message = "User ID must not be null")
    private Long userId;
    @NotNull(message = "Contact must not be null")
    @Valid
    private ContactRequestDto contact;
    @Valid
    private List<EducationRequestDto> educations = new ArrayList<>();
    @Valid
    private List<ExperienceRequestDto> experiences = new ArrayList<>();
    @Valid
    private List<LanguageRequestDto> languages = new ArrayList<>();
    @Valid
    private List<TechnicalSkillRequestDto> technicalSkills = new ArrayList<>();
    @Valid
    private List<CertificationRequestDto> certifications = new ArrayList<>();
    @Valid
    private List<ContactRequestDto> references = new ArrayList<>();
    @Valid
    private List<ResumeFileRequestDto> resumeFiles = new ArrayList<>();
    @NotNull(message = "Is current must not be null")
    private Boolean isCurrent;
}

File: ungs-docsys-front/src/commons/dtos/resume-user-request.dto.ts
import { CertificationRequestDto } from './certification-request.dto';
import { ContactRequestDto } from './contact-request.dto';
import { EducationRequestDto } from './education-request.dto';
import { ExperienceRequestDto } from './experience-request.dto';
import { LanguageRequestDto } from './language-request.dto';
import { ResumeFileRequestDto } from './resume-file-request.dto';
import { TechnicalSkillRequestDto } from './technical-skill-request.dto';

export class ResumeUserRequestDto {
  contact: ContactRequestDto;
  educations: EducationRequestDto[] = [];
  experiences: ExperienceRequestDto[] = [];
  languages: LanguageRequestDto[] = [];
  technicalSkills: TechnicalSkillRequestDto[] = [];
  certifications: CertificationRequestDto[] = [];
  references: ContactRequestDto[] = [];
  resumeFiles: ResumeFileRequestDto[] = [];
}